controller/array_action/svc_cli_result_reader.py

```diff
@@ -0,0 +1,117 @@
+from controller.common.csi_logger import get_stdout_logger
+import controller.array_action.errors as controller_errors
+
+logger = get_stdout_logger()
+
+ID_PARAM_NAME = "id"
+
+
+class SVCListResultsReader:
+    """
+    Iterable object used to read raw command response from SVC array.
+    Each object in response is translated to SVCListResultsElement.
+    Input is received as string which represents '\n'-separated list of returned lines from output.
+    Line with param 'id' (e.g. 'id 1') is recognized as first line of object ans used as separator between objects
+    (e.g. in input "id 1\nname n3\nid 2\nname n2" first object starts with line 'id 1' and ends with 'id 2')
+    """
+    def __init__(self, hosts_raw_list_as_string):
+        self._hosts_raw_list = hosts_raw_list_as_string.splitlines()
+        self._current_index = 0
+        self._next_object_id = None
+        self._init_first_object_id()
+
+    def _init_first_object_id(self):
+        """
+        Set _init_first_object_id to id o the first object and _current_index to point to next line
+
+        Raises:
+            InvalidCliResponseError
+        """
+        while not self._next_object_id and self._current_index < len(self._hosts_raw_list):
+            line = self._hosts_raw_list[self._current_index].strip()
+            self._current_index += 1
+            if line:
+                param_name, param_value = self._parse_param(line)
+                if param_name == ID_PARAM_NAME:
+                    self._next_object_id = param_value
+                else:
+                    raise controller_errors.InvalidCliResponseError(
+                        "First element is {0}. Expected {1}".format(line, ID_PARAM_NAME))
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        """
+        Assumed self._current_index points to the line after line 'id <id>' of the next object
+
+        Returns:
+            Next object as SVCListResultsElement.
+        Raises:
+            StopIteration
+        """
+
+        if not self._has_next():
+            raise StopIteration
+
+        res = SVCListResultsElement()
+        res.add(ID_PARAM_NAME, self._next_object_id)
+        self._next_object_id = None
+        while self._current_index < len(self._hosts_raw_list):
+            line = self._hosts_raw_list[self._current_index].strip()
+            self._current_index += 1
+            if not line:
+                continue
+            param_name, param_value = self._parse_param(line)
+            if param_name == ID_PARAM_NAME:
+                self._next_object_id = param_value
+                return res
+            res.add(param_name, param_value)
+        return res
+
+    def _parse_param(self, line):
```

Comment:
all this function for `name, _, value = line.partition(' ')`? what am I missing here?
you can also add `value = value.strip()`, but I am not aware of such need

Reply:
1) I don't assume space separator but and whitespace
2) For empty value there ay be only single sting e.g. line "iscsi_name iscsi.ibm " or "iscsi_name" if no iscsi
3) value can contain spaces

Reply:
I don't think 1. is a real scenario, but I'm pretty sure my suggestion works with 2. + 3...

Reply:
done